File: Book_search/app/src/main/java/com/example/book_search/LoginActivity.java
package com.example.book_search;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.content.Intent;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        EditText username =  findViewById(R.id.usrnametext);
        EditText password =  findViewById(R.id.passtext);

        Button loginbutton =  findViewById(R.id.Signinbtn);

        loginbutton.setOnClickListener(view -> {

            if(username.getText().toString().trim().length()==0||password.getText().toString().trim().length()==0)
            {
                Toast.makeText(LoginActivity.this, getString(R.string.login_failed),
                        Toast.LENGTH_SHORT).show();
            }
            else {
                Toast.makeText(LoginActivity.this, getString(R.string.login_successful),Toast.LENGTH_SHORT).show();
                String uname=username.getText().toString();
                Intent intent = new Intent(LoginActivity.this,WelcomeUserActivity.class);
                intent.putExtra(getString(R.string.username),uname);
                startActivity(intent);
            }

        });
    }
}

File: app/src/main/java/com/example/book_search_kotlin/LoginActivity.kt
package com.example.book_search_kotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val usernameText : EditText = findViewById(R.id.usernameText)
        val passwordText : EditText = findViewById(R.id.passwordText)
        val loginButton  : Button = findViewById(R.id.loginbutton)

        loginButton.setOnClickListener(){
            val login = Login()
            val key: Boolean=login.LoginAuth(usernameText.text.toString(),passwordText.text.toString())
            if(key)
            {
                val intent =Intent(this@LoginActivity,WelcomeActivity::class.java)
                intent.putExtra("userName",usernameText.text.toString())
                startActivity(intent)
            }
            else
            {
                Toast.makeText(this@LoginActivity,"Login Failed check credentials",Toast.LENGTH_SHORT).show()
            }


        }

    }
}